src/libPersistence/IncrementalDB.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2019 Zilliqa
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#ifndef __INRCREMENTAL_DB_H__
+#define __INRCREMENTAL_DB_H__
+
+#include <boost/filesystem.hpp>
+#include <list>
+#include <mutex>
+#include <shared_mutex>
+#include <vector>
+
+#include "BlockStorage.h"
+#include "common/Singleton.h"
+#include "depends/libDatabase/LevelDB.h"
+#include "libData/BlockData/Block.h"
+#include "libData/BlockData/Block/FallbackBlockWShardingStructure.h"
+#include "libValidator/Validator.h"
+
+class IncrementalDB : public Singleton<IncrementalDB> {
+  std::unordered_map<std::string, std::pair<uint64_t, std::shared_ptr<LevelDB>>>
```

Comment:
I suggest not to save all the leveldb instance in this in-memory map, should replace or clean the earlier leveldb instance for every ds epoch being finished

Reply:
it is not saved, when there is no pointer pointing to it, it is already cleared